src/htmx.js

```diff
@@ -5121,12 +5121,13 @@ var htmx = (function() {
  */
 
 /**
+ * @see https://github.com/bigskysoftware/htmx-extensions/blob/main/README.md
  * @typedef {Object} HtmxExtension
- * @see https://htmx.org/extensions/#defining
  * @property {(api: any) => void} init
  * @property {(name: string, event: Event|CustomEvent) => boolean} onEvent
  * @property {(text: string, xhr: XMLHttpRequest, elt: Element) => string} transformResponse
  * @property {(swapStyle: HtmxSwapStyle) => boolean} isInlineSwap
- * @property {(swapStyle: HtmxSwapStyle, target: Element, fragment: Node, settleInfo: HtmxSettleInfo) => boolean} handleSwap
- * @property {(xhr: XMLHttpRequest, parameters: FormData, elt: Element) => *|string|null} encodeParameters
+ * @property {(swapStyle: HtmxSwapStyle, target: Node, fragment: Node, settleInfo: HtmxSettleInfo) => boolean} handleSwap
```

Comment:
Hey, good catch!
You'll probably want to update the JSDoc of `handleSwap` to indicate that double return type then (only specifies a `boolean` as return type atm), as tsc currently fails the types generation step (see the errors reported by the CI above)

Reply:
Thanks! That was basically what I was asking for, and allowing both types is for sure the best option!

Unfortunately, TSC seems to not properly handle the `typeof newElements.length !== 'undefined'` as it still throws the same type error. I therefore changed the condition to a `Array.isArray` call, which is widely available in all Browsers: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray